packages/shared/src/hooks/useElementOnScreen.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useRef } from 'react'
+
+export const useElementOnScreen = (callback: (isIntersecting: boolean) => void, option: any) => {
```

Comment:
I suggest installing `react-hookz`, which is created by the author of `react-use` (react-use is stop maintaining).
<https://react-hookz.github.io/web/?path=/docs/sensor-useintersectionobserver--example>